app/src/main/AndroidManifest.xml

```diff
@@ -23,13 +22,15 @@
     <uses-permission android:name="android.permission.RECEIVE_SMS" />
     <uses-permission android:name="android.permission.READ_SMS" />
 
+    <android:uses-permission android:name="android.permission.READ_PHONE_STATE" />
+    <android:uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```

Comment:
Why the android prefix? Android studio is complaining. Why so much carelessness?